src/main/java/com/redhat/coolstore/service/OrderService.java

```diff
@@ -1,33 +1,41 @@
 package com.redhat.coolstore.service;
 
-import com.redhat.coolstore.model.Order;
-import java.util.List;
-import javax.ejb.Stateless;
+import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
-import javax.persistence.EntityManager;
-import javax.persistence.criteria.CriteriaBuilder;
-import javax.persistence.criteria.CriteriaQuery;
-import javax.persistence.criteria.Root;
+import io.quarkus.hibernate.orm.panache.PanacheQuery;
+import io.quarkus.hibernate.orm.panache.PanacheRepository;
+import io.quarkus.panache.common.Page;
+import io.quarkus.panache.common.Sort;
 
-@Stateless
+@ApplicationScoped
 public class OrderService {
 
-  @Inject
-  private EntityManager em;
+    @Inject
+    OrderRepository orderRepository;
 
-  public void save(Order order) {
-    em.persist(order);
-  }
+    public PanacheQuery<Order> findAllOrders(Integer page, Integer size) {
```

Comment:
Again adding some Panache stuff to simplify the code. It kind of makes sense, although we would have to check if it's spot on. In any case, none of this is in our rules.